src/backends/tensorrt/plan_backend.cc

```diff
@@ -843,11 +843,20 @@ PlanBackend::Context::InitializeShapeInputBinding(
   }
 
   if (max_byte_size != NO_BATCHING) {
-    // Allocate CUDA memory. We rely on buffer_bindings_ being non-nullptr to
-    // indicate that the buffer has been correctly initalized so even
-    // for zero-sized tensors always allocate something.
+    // Allocate CUDA memory. Use cudaHostAlloc if zero copy supported. We rely
+    // on buffer_bindings_ being non-nullptr to indicate that the buffer has
+    // been correctly initalized so even for zero-sized tensors always allocate
+    // something.
+    bool zero_copy_support = false;
     void* buffer;
-    cudaError_t err = cudaMalloc(&buffer, std::max((int64_t)1, max_byte_size));
+    cudaError_t err;
+    Status status = CheckGPUZeroCopySupport(gpu_device_, &zero_copy_support);
```

Comment:
This property doesn't change so we should just be able to calculate it once when the instance is created

Reply:
I keep this as it is instead of storing this property as class member because this property is checked only when creating the I/O buffer and not used afterwards.